code/obj/item/device/radios/headsets.dm

```diff
@@ -17,38 +17,62 @@
 	wear_layer = MOB_EARS_LAYER
 	duration_remove = 1.5 SECONDS
 	duration_put = 1.5 SECONDS
-	var/haswiretap
+	var/obj/item/device/radio_upgrade/wiretap = null
```

Comment:
(for the entire code)
tbh it could do better with being less snowflakey by making the radio upgrades into a list, and then giving the headset channels from a list *on* the radio upgrade itself (or in special snowflake cases like the conspiracy channel, a proc on the headset)